src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -576,6 +576,12 @@ private object GetVSSolutionProperty(int propId)
         {
             Debug.Assert(ThreadHelper.CheckAccess());
 
+            // if VSSolutionManager Initialization fail, _vsSolution can be null.
+            if (_vsSolution == null)
+            {
+                throw new ArgumentNullException(nameof(_vsSolution));
```

Comment:
This indicates the object is in invalid state. Throw `InvalidOperationException` with error message instead.